Assets/00_Content/Scripts/Interactables/Instruments/Instrument.cs

```diff
@@ -97,10 +117,19 @@ public class Instrument : PickUp, IUseable, IDesirable {
 
 			Viking viking = other.attachedRigidbody.GetComponent<Viking>();
 
-			if (viking != null) {
-				vikingsInRange.Remove(viking);
-				if (isPlaying)
-					viking.CancelAffect(usingPlayer, this);
+			if (viking == null) return;
+
+			if (vikingsInRange.ContainsKey(viking)) {
+				int value = --vikingsInRange[viking];
+				if (value <= 0) {
+					vikingsInRange.Remove(viking);
```

Comment:
This is scary, would having a separate layer (I know, so many layers...) for instrument AoE to collide with vikings only be nicer? It seems to collide with a viking's all different colliders

Reply:
Not any scarier than what we had before. It's basically doing the same thing but storing it differently.
But yes, I thought about that, but like... layers... I would probably need to add 2 new layers in that case. I also thought about tags, but it seemed like that would also be pretty nasty.